### PR TITLE
Rebase main

### PR DESCRIPTION
**Pull request by nelsonand**
This starts a pull request to rebase the working function into a single main script. 

Options of more to do:
 - use a class format instead?
 - separate out more functionality into dedicated functions?

@namurphy do you have suggestions here?

### CODE REVIEW

File: scripts/analysis/main.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+import git
+import pandas as pd
+import os
+import json
+import numpy as np
+
+# establish relative directories
+repo = git.Repo('.', search_parent_directories=True)
+os.chdir(repo.working_tree_dir)
```
Comment (nelsonand) [inline scripts/analysis/main.py:19]: Is there a smarter way to handle directories?

Comment (namurphy) [reply scripts/analysis/main.py:19]: Perhaps use [`pathlib`](https://docs.python.org/3/library/pathlib.html), and in particular `pathlib.Path`?

File: scripts/analysis/main.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        compscore = (nNorm-(sdata.Rank-1)) / (nNorm/10) # normalized ranking (0-10)
+        sdata['CompScore'] = compscore
+    elif 'Processed' in sdata.keys():
+        sdata['CompScore'] = sdata.Processed
+    else:
+        print(f' ---   WARNING: unclear interpretation of CompScore for {source}... this will break!') 
+
+    return sdata
+
+def nornalize_CompScore(sdata):
```
Comment (namurphy) [inline scripts/analysis/main.py:213]:
```suggestion
def normalize_CompScore(sdata):
```

File: scripts/analysis/main.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+import json
+import numpy as np
+
+# establish relative directories
+repo = git.Repo('.', search_parent_directories=True)
+os.chdir(repo.working_tree_dir)
+
+def gardnm(
+    cities_to_print = ['Atlanta', 'Boston', 'Memphis'],  
+    noramlizeAll = True,
```
Comment (namurphy) [inline scripts/analysis/main.py:23]:
```suggestion
    normalizeAll = True, 
```

File: scripts/analysis/main.py [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    filename = 'gardnm', 
+    oldPSW = False, 
+    setPbyCity = True,   
+    verbose = False, 
+):
+    """Calculation of GARDN-M coefficients
+
+    Args:
+        cities_to_print: print the output of these cities to the console 
+        noramlizeAll: normalize raw data from every source to span the full 0-10 scale
```
Comment (namurphy) [inline scripts/analysis/main.py:33]:
```suggestion
        normalizeAll: normalize raw data from every source to span the full 0-10 scale
```

**Pull request merged**